src/tribler-core/tribler_core/components/restapi/restapi_component.py

```diff
@@ -85,36 +65,63 @@ async def run(self):
         notifier = session.notifier
         shutdown_event = session.shutdown_event
 
-        root_endpoint = RootEndpoint(config, middlewares=[ApiKeyMiddleware(config.api.key), error_middleware])
-        rest_manager = RESTManager(config=config.api, root_endpoint=root_endpoint, state_dir=config.state_dir)
-        # Unfortunately, AIOHTTP endpoints cannot be added after the app has been started.
-        # On the other hand, we have to start the state endpoint from the beginning, to
-        # communicate with the upgrader. Thus, we start the endpoints immediately and
-        # then gradually connect them to their respective backends during the core start process.
+        log_dir = config.general.get_path_as_absolute('log_dir', config.state_dir)
+        metadata_store_component = await self.get_component(MetadataStoreComponent)
+
+        # fmt: off
+        # pylint: disable=C0301
+        key_component                  = await self.require_component(KeyComponent)
+        ipv8_component                 = await self.maybe_component(Ipv8Component)
+        libtorrent_component           = await self.maybe_component(LibtorrentComponent)
+        resource_monitor_component     = await self.maybe_component(ResourceMonitorComponent)
+        bandwidth_accounting_component = await self.maybe_component(BandwidthAccountingComponent)
+        gigachannel_component          = await self.maybe_component(GigaChannelComponent)
+        tag_component                  = await self.maybe_component(TagComponent)
+        tunnel_component               = await self.maybe_component(TunnelsComponent)
+        torrent_checker_component      = await self.maybe_component(TorrentCheckerComponent)
+        gigachannel_manager_component  = await self.maybe_component(GigachannelManagerComponent)
```

Comment:
**Note:** this formatting is nice, but it will be reformated with high probability by IDE on the next change.

Reply:
That's why I put `fmt: off` there.

Reply:
FYI: https://stackoverflow.com/questions/3375307/how-to-disable-code-formatting-for-some-part-of-the-code-using-comments

Reply:
> That's why I put fmt: off there.

It will not help with IDE formatting.

Reply:
> It will not help with IDE formatting.

If you enable your IDE to do so (see the link I posted above), it will. Remind that we don't have an official guideline for the code to be friendly to a particular IDE-based formatting tool. Instead, we have the `black` pre-commit hook (which respects `fmt: off`).

Reply:
Black pre-commit hook is not mandatory to use nor agreed to use by all the team.

I just noticed here, that a person who will change a line in this particular file will remove all formatting with a very high probability.

Reply:
> will remove all formatting with a very high probability.

I'm fine with that. That's for that person and their PR reviewer to decide what to do with the code they change :wink: